### PR TITLE
v1.6.3 release

### PR DESCRIPTION
**Pull request by creisle**
New features
- ymax and stranded options for overlay diagram read depth plots
- clean up option for aligner output files
- write option for outputting evidence bams, bed files etc (can be turned off now)

Improvements
- Producing products for exact calls where untemplated sequence was not resolved by assuming it is empty
- Caching less transcriptome reads to reduce memory footprint

Bugfixes
- (minor) improved resolve by split reads, previously was not resolving unstranded calls because it was assuming the strands represented different resolutions

### CODE REVIEW

File: mavis/annotate/variant.py [hunk 1]
```diff
@@ -710,28 +710,34 @@ def annotate_events(
...
             proximity=max_proximity
         )
         for f in filters:
-            ann = f(ann)  # apply the filter
-        results.extend(ann)
-        for j, a in enumerate(ann):
-            a.data[COLUMNS.annotation_id] = '{}-a{}'.format(a.validation_id, j + 1)
+            ann_list = f(ann_list)  # apply the filter
+        results.extend(ann_list)
+        for j, ann in enumerate(ann_list):
```
Comment (dpaulino) [inline mavis/annotate/variant.py]: Perhaps I am just a bit unfamiliar with this, but what is `j` here? Can another name be used to clarify it?

Comment (dustinbleile) [reply mavis/annotate/variant.py]: 'j' is the index of the enumerated ann_list, like 'i' is for bpps.
The 'i', 'j', 'k', ... indexing convention is very old.  Comes from FORTRAN variable naming rules.

File: mavis/illustrate/scatter.py [hunk 2]
```diff
@@ -1,4 +1,73 @@
...
+    """
+    import pysam
+    if not axis_name:
+        axis_name = os.path.basename(bam_file)
+    # one plot per bam
+    log('reading:', bam_file)
+    plot = None
+    samfile = pysam.AlignmentFile(bam_file, 'rb')
+
+    def read_filter(read):
```
[the remark this answers is not in the document]
Comment (creisle) [reply mavis/illustrate/scatter.py]: that is documented on the function that this is being passed into

File: mavis/main.py [hunk 3]
```diff
@@ -300,28 +301,40 @@ def parse_overlay_args(parser, required, optional):
...
             parser.error('argument --marker: start and end must be integers: {}'.format(marker))
 
+    defaults = [None, None, 1, None, True]
     for plot in args.read_depth_plots:
-        if len(plot) < 3:
-            plot.append(1)
+        for i, d in enumerate(defaults):
+            if i >= len(plot):
+                plot.append(d)
         if not os.path.exists(plot[1]):
```
[the remark this answers is not in the document]
Comment (creisle) [reply mavis/main.py]: b/c I am modifying the array in place so all the assignments need to be by index

File: mavis/validate/call.py [hunk 4]
```diff
@@ -846,45 +846,66 @@ def _call_by_supporting_reads(evidence, event_type, consumed_evidence=None):
...
+                call.add_flanking_support(available_flanking_pairs)
+                if call.has_compatible:
+                    call.add_flanking_support(available_flanking_pairs, is_compatible=True)
+                # add the initial reads
+                for read in uncons_break1_reads - consumed_evidence:
+                    call.add_break1_split_read(read)
+                for read in uncons_break2_reads - consumed_evidence:
+                    call.add_break2_split_read(read)
+                # does it pass the requirements?
+                if any([
```
[the remark this answers is not in the document]
Comment (creisle) [reply mavis/validate/call.py]: I don't agree with if/else over indenting all the time but I think you're right in this case. I like this answer: https://softwareengineering.stackexchange.com/questions/58237/are-break-and-continue-bad-programming-practices

File: mavis/main.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dustinbleile) [reply mavis/main.py]: BAM_FILE = 1
plot[BAM_FILE] = xxx

is still assignment by index.  Why not use meaningful constants for the index values?

[the remark this answers is not in the document]
Comment (creisle) [reply mavis/main.py]: ah, sorry I misread what you wrote above

File: mavis/validate/call.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (creisle) [reply mavis/validate/call.py]: this actually brings up a bigger issue of me filtering out insertions here, should be checking flanking pairs for those as well. Also should check for non-forced reads so will add those 3 lines here.

File: mavis/annotate/variant.py (hunk 1 above)
Comment (dpaulino) [reply mavis/annotate/variant.py]: Oh I understand! I'm wasn't very familiar with how enumerate() works in python for loops, so I forgot the first value is an index. Thanks for taking the time to clarify that, Dustin.

**Pull request merged**